src/backend/tty.rs

```diff
@@ -1282,6 +1288,54 @@ impl Tty {
             return;
         };
 
+        if let Some(vblank_throttle_timer_token) = output_state.vblank_throttle_timer_token.take() {
+            niri.event_loop.remove(vblank_throttle_timer_token);
+        }
+
+        let vblank_remaining_time =
+            output_state
+                .last_drm_vblank_timestamp
+                .and_then(|last_drm_vblank_timestamp| {
+                    let refresh_interval = output_state.frame_clock.refresh_interval()?;
+                    let vblank_diff = time.saturating_sub(last_drm_vblank_timestamp);
+                    Some(refresh_interval.saturating_sub(vblank_diff))
+                });
+
+        if let Some(vblank_remaining_time) = vblank_remaining_time.filter(|vblank_remaining_time| {
+            output_state
+                .frame_clock
+                .refresh_interval()
```

Comment:
nit: might be cleaner to extract `if let Some(refresh_interval) = output_state.frame_clock.refresh_interval() {` outside this whole vblank_remaining_time handling section